app/src/main/java/com/keylesspalace/tusky/fragment/SFragment.java

```diff
@@ -269,10 +274,17 @@ private static boolean looksLikeMastodonUrl(String urlString) {
             return false;
         }
 
-        return (uri.getQuery() == null &&
-                uri.getFragment() == null &&
-                uri.getPath() != null &&
-                uri.getPath().matches("^/(@|notice).*/\\d+$"));
+        if (uri.getQuery() != null ||
+                uri.getFragment() != null ||
+                uri.getPath() == null) {
+            return false;
+        }
+
+        String path = uri.getPath();
+        return path.matches("^/@[^/]*$") ||
```

Comment:
I think it could be more efficient to smash them all into one regex but I won't be able to do such wizardry nor I do I ask you, just saying

Reply:
I could make them a single regex, but I thought this was more readable/debuggable. The time taken to match against all the regexes in the worst case is still tiny compared to the actual network request (either for the search or for the browser).

Reply:
Yeah, you're right.